package.json

```diff
@@ -116,5 +116,19 @@
   "packageManager": "npm@8.19.4",
   "engines": {
     "node": "14 >=14.21 || 16 >=16.20 || 18 || 20 || >=22"
+  },
+  "devEngines": {
+    "runtime": [
+      {
+        "name": "node",
+        "version": "^16.20"
+      }
+    ],
+    "packageManager": [
+      {
+        "name": "npm",
+        "version": "^8.19.4"
```

Comment:
Interesting!

Reply:
Did realize that there's some bugs and uncompleted functionality surrounding this, so I might revert this for now, unfortunately (npm/cli#7888 & commentary in npm/cli#7766).